### PR TITLE
Pancakes column names

### PR DESCRIPTION
**Pull request by jakozaur**
Pancakes depend on unique names. The code needs to deal with edge cases well.

### CODE REVIEW

File: quesma/queryparser/pancake_transformer.go [hunk 1]
```diff
@@ -6,12 +6,43 @@ import (
...
 type pancakeTransformer struct {
+	usedNames map[string][]string
+}
+
+// Extremely rarely names may collide (e.g. aggregation named '1__2' and '1', '2').
+// This adds number suffix to make sure they are always unique
+func (a *pancakeTransformer) generateUniqueInternalName(origName string, aggrNames []string) string {
+	if a.usedNames == nil {
+		a.usedNames = make(map[string][]string)
+	}
```
Comment (trzysiek) [inline quesma/queryparser/pancake_transformer.go:24]:
```suggestion
	usedNames map[string][]string
}

func newPancakeTransformer() pancakeTransformer {
    return pancakeTransformer{make(map[string][]string}
}

// Extremely rarely names may collide (e.g. aggregation named '1__2' and '1', '2').
// This adds number suffix to make sure they are always unique
func (a *pancakeTransformer) generateUniqueInternalName(origName string, aggrNames []string) string {
```

Comment (trzysiek) [inline quesma/queryparser/pancake_transformer.go:24]:
```suggestion
func newPancakeTransformer() pancakeTransformer {
    return pancakeTransformer{make(map[string][]string}
}

// Extremely rarely names may collide (e.g. aggregation named '1__2' and '1', '2').
// This adds number suffix to make sure they are always unique
func (a *pancakeTransformer) generateUniqueInternalName(origName string, aggrNames []string) string {
```

Comment (trzysiek) [reply quesma/queryparser/pancake_transformer.go:24]: Nitpick, but I'd change it to this, and use this constructor. This way we don't need to worry about concurrency in this `nil` check.

**Review by trzysiek — APPROVED** [review]
easy and clean implementation, I like it!

File: quesma/queryparser/pancake_transformer.go (hunk 1 above)
Comment (jakozaur) [reply quesma/queryparser/pancake_transformer.go:24]: Ok, will do.

**Pull request merged**